rb/lib/selenium/webdriver/common/interactions/pointer_actions.rb

```diff
@@ -0,0 +1,359 @@
+      #   that will be pressed, defaults to the primary PointerInput device.
+      # @return [W3CActionBuilder] A self reference.
+      #
+
+      def pointer_down(button, device_name = nil)
+        pointer = get_device(device_name) || primary_pointer
+        pointer.create_pointer_down(button)
+        tick(pointer)
+        self
+      end
+
+      #
+      # Releases the pressed mouse button at the current mouse location of the PointerInput device.
+      #
+      # @example Releasing a button after clicking and holding
+      #
+      #    driver.action.pointer_down(:left).pointer_up(:left).perform
+      #
+      # @param [Selenium::WebDriver::Interactions::PointerPress::BUTTONS] button the button to release.
+      # @param [Symbol || String] device_name optional name of the PointerInput device with the button that will
+      #   be released, defaults to the primary PointerInput device.
+      # @return [W3CActionBuilder] A self reference.
+      #
+
+      def pointer_up(button, device_name = nil)
+        pointer = get_device(device_name) || primary_pointer
+        pointer.create_pointer_up(button)
+        tick(pointer)
+        self
+      end
+
+      #
+      # Moves the mouse to the middle of the given element. The element is scrolled into
+      # view and its location is calculated using getBoundingClientRect.  Then the
+      # mouse is moved to optional offset coordinates from the element.
+      #
+      # This is adapted to be backward compatible from non-W3C actions. W3C calculates offset from the center point
+      # of the element
+      #
+      # Note that when using offsets, both coordinates need to be passed.
+      #
+      # @example Scroll element into view and move the mouse to it
+      #
+      #   el = driver.find_element(id: "some_id")
+      #   driver.action.move_to(el).perform
+      #
+      # @example
+      #
+      #   el = driver.find_element(id: "some_id")
+      #   driver.action.move_to(el, 100, 100).perform
+      #
+      # @param [Selenium::WebDriver::Element] element to move to.
+      # @param [Integer] right_by Optional offset from the top-left corner. A negative value means
+      #   coordinates to the left of the element.
+      # @param [Integer] down_by Optional offset from the top-left corner. A negative value means
+      #   coordinates above the element.
+      # @param [Symbol || String] device_name optional name of the PointerInput device to move.
+      # @return [W3CActionBuilder] A self reference.
+      #
+
+      def move_to(element, right_by = nil, down_by = nil, device_name: nil)
+        pointer = get_device(device_name) || primary_pointer
+        # New actions offset is from center of element
+        if right_by || down_by
+          size = element.size
+          left_offset = (size[:width] / 2).to_i
+          top_offset = (size[:height] / 2).to_i
+          left = -left_offset + right_by
+          top = -top_offset + down_by
```

Comment:
I put it this way to show the steps of calculating the final value. Center point - left/top offset in order to get 0(ish) location, then applying the user's offset.